src/gc-stacks.c

```diff
@@ -215,33 +216,38 @@ void sweep_stack_pools(void) JL_NOTSAFEPOINT
     //            bufsz = t->bufsz
     //            if (stkbuf)
     //                push(free_stacks[sz], stkbuf)
-    assert(gc_n_threads);
-    for (int i = 0; i < gc_n_threads; i++) {
+    jl_atomic_fetch_add(&gc_n_threads_sweeping, 1);
```

Comment:
Similar comment as before: using a new counter `gc_n_threads_sweeping_stacks` instead of re-using the old `gc_n_threads_sweeping` makes the code easier to reason about here.